lib/msf/core/payload/adapter/fetch/server/https.rb

```diff
@@ -0,0 +1,148 @@
+      fail_with(Msf::Exploit::Failure::BadConfig, "Called an HTTPS method after importing the HTTP mixin")
+    end
+  end
+
+  # We likely don't need these ssl_check calls in these methods, but it will help debugging
+  #
+  def ssl
+    check_ssl
+    return datastore['FETCH_SSL']
+    false
+  end
+
+  def ssl_cert
+    check_ssl
+    datastore['FetchSSLCert']
+  end
+
+  def ssl_compression
+    check_ssl
+    datastore['FetchSSLCompression']
+  end
+
+  def ssl_cipher
+    check_ssl
+    datastore['FetchSSLCipher']
+  end
+
+  def ssl_version
+    check_ssl
+    datastore['FetchSSLVersion']
+  end
+
+
+  def start_http_fetch_handler(srvname, srvexe)
+    start_https_fetch_handler(srvname, srvexe)
+  end
+
+  def start_https_server(ssl, ssl_cert, ssl_compression, ssl_cipher, ssl_version)
+    begin
+      fetch_service = Rex::ServiceManager.start(
+        Rex::Proto::Http::Server,
+        fetch_bindport, fetch_bindhost, ssl,
+        {
+          'Msf' => framework,
+          'MsfExploit' => self
+        },
+        _determine_server_comm(fetch_bindhost),
+        ssl_cert,
+        ssl_compression,
+        ssl_cipher,
+        ssl_version
+      )
+    rescue Exception => e
+      cleanup_handler
+      fail_with(Msf::Exploit::Failure::BadConfig, "Fetch Handler failed to start on #{srvhost}:#{srvport}\n #{e}")
+    end
+    print_status("Fetch Handler listening on #{srvhost}:#{srvport}")
+    fetch_service
+  end
+
+  def start_https_fetch_handler(srvname, srvexe)
+    # We assume that some options will be present in the datastore if we are here....
+    check_ssl
+    fetch_service = start_https_server(ssl, ssl_cert, ssl_compression, ssl_cipher, ssl_version)
+    if fetch_service.nil?
+      cleanup_handler
+      fail_with(Msf::Exploit::Failure::BadConfig, "Fetch Handler failed to start on #{srvhost}:#{srvport}\n #{e}")
+    end
+    vprint_status('http[s] server started')
+    vprint_status("http[s] server name: #{fetch_service.server_name}")
+    fetch_service.server_name = srvname
+    vprint_status('http[s] server name changed')
+    fetch_service.add_resource(('/' + srvuri).gsub('//', '/'),
+                               'Proc' => proc do |cli, req|
+                                 on_request_uri(cli, req, srvexe)
+                               end,
+                               'VirtualDirectory' => true)
+    vprint_status("Added resource #{srvuri}")
+    fetch_service
+  end
+
+  def payload_response(srvexe)
+    res = Rex::Proto::Http::Response.new(200, 'OK', Rex::Proto::Http::DefaultProtocol)
+    res['Content-Type'] = 'text/html'
+    res.body = srvexe.to_s.unpack('C*').pack('C*')
+    res
+  end
+
+  def cleanup_http_fetch_service
+    vprint_status('Trying to clean up fetch http service')
+    unless @fetch_service.nil?
+      vprint_status("#{@fetch_service.resources}")
+      @fetch_service.remove_resource('/' + srvuri) unless @fetch_service.nil?
+      vprint_status("#{@fetch_service.resources}")
```

Comment:
It looks like these two `vprint_status` are leftover from development debugging. I'm wondering if this is useful from the end user to have this information.